index.js

```diff
@@ -87,10 +88,23 @@ function requestAsEventEmitter(opts) {
 }
 
 function asPromise(opts) {
-	return new Promise((resolve, reject) => {
+	return new PCancelable((onCancel, resolve, reject) => {
 		const ee = requestAsEventEmitter(opts);
+		let cancelImmediately = false;
+
+		onCancel(() => {
+			cancelImmediately = true;
+		});
 
 		ee.on('request', req => {
+			if (cancelImmediately) {
+				req.abort();
+			}
+
+			onCancel(() => {
+				req.abort();
+			});
```

Comment:
Hmmm. Alright. That means coming up with a way to easily test the request actually gets canceled. Don't want to simply repeat what I did in the previous test. I'll come up with something clean. Give me a couple minutes.